### PR TITLE
RFC-0030: FP8 dtype introduction to PyTorch

### PR DESCRIPTION
**Pull request by australopitek**
This RFC proposes adding 8-bit floating point data types to PyTorch.

### CODE REVIEW

File: RFC-0030-native-fp8-dtype.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+* @australopitek
+
+
+## **Summary**
+More and more companies working on Deep Learning accelerators are experimenting with 8-bit floating point numbers usage in training and inference. Results of these experiments are presented in many papers published in the last few years.
+
+Since fp8 data type seems to be a natural evolution of currently used fp16/bf16, to reduce computation of big DL models, it’s worth to standardize this type. Few attempts of this were done recently:
+
+* Nvidia, Arm and Intel - https://arxiv.org/pdf/2209.05433.pdf
+* GraphCore and AMD - https://arxiv.org/pdf/2206.02915.pdf
```
Comment (jakeh-gc) [inline RFC-0030-native-fp8-dtype.md:13]: For completeness, these formats are proposed by Graphcore, AMD, and Qualcomm.

Comment (australopitek) [reply RFC-0030-native-fp8-dtype.md:13]: I'll correct it when more comments are there.

**Comment by baobablyh** [comment]
Curious what the progress for fp8 support looks like? Thanks!

**Comment by jakeh-gc** [comment]
@australopitek I've been working more on the XLA side. The only activity I've seen in PyTorch was this https://github.com/pytorch/pytorch/pull/97798, which didn't get merged.

**Comment by albanD** [comment]
Hey!
Wrote an update at https://github.com/pytorch/pytorch/issues/91577#issuecomment-1593172800 if you want to comment there!

**Comment by timljj** [comment]
Hi @australopitek, in your md file, you mentioned that for E5M2 "there are many models that can be trained only with this variant". 

May I know what models/type of models you are referring to? 

Also, does your statement mean that those models would not be able to be trained with E4M3?

**Comment by australopitek** [comment]
Hi @timljj ,
my wording was not precise. I wanted to say that some models can be trained with E5M2 without "help" of E4M3, i.e. E5M2 is sufficient (despite being "Applicable mainly for gradients in the backward pass of training").
Such experiments are described in the https://arxiv.org/pdf/1905.12334.pdf - Resnet18, Resnet34, Resnet50, GNMT, Transformer.

**Comment by maxpain** [comment]
Any updates?

**Comment by australopitek** [comment]
@maxpain,
float8 dtypes are already delivered to pytorch - https://github.com/pytorch/pytorch/pull/104242 - I guess they will be released in PT2.1
@albanD,
should we commit this RFC for the sake of completeness?
